src/hotspot/share/runtime/thread.cpp

```diff
@@ -482,6 +482,27 @@ void Thread::check_for_dangling_thread_pointer(Thread *thread) {
 }
 #endif
 
+// Is the target JavaThread protected by this Thread:
+bool Thread::is_JavaThread_protected(const JavaThread* p) {
```

Comment:
I made it a function on Thread because a Thread can use a ThreadsListHandle
to protect access to JavaThreads and we're checking the ThreadsList(s) that
are associated with the calling Thread.

It dawns on me that the comment could have been more clear:

`// Is the target JavaThread protected by the calling Thread:`

If I had written the comment that way, then I would have realized that the
function should be static in Thread and then it could call Thread::current()
itself and then make its checks of the ThreadsList(s).